<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <!-- Run keyboard.launch from another terminal window -->

    <arg name="image_throttle_rate" default="1.0"/>

    <!-- <node type="throttle" name="image_topic_throttle" pkg="topic_tools" required="true" output="screen"
        args="messages /camera/color/image_raw $(arg image_throttle_rate) /camera/color/image_raw_throttled"/> -->
    <node type="throttle" name="depth_topic_throttle" pkg="topic_tools" required="true" output="screen"
        args="messages /camera/aligned_depth_to_color/image_raw $(arg image_throttle_rate) /camera/aligned_depth_to_color/image_raw_throttled"/>
    <!-- <node type="throttle" name="image_info_topic_throttle" pkg="topic_tools" required="true" output="screen"
        args="messages /camera/color/camera_info $(arg image_throttle_rate) /camera/color/camera_info_throttled"/> -->
    <!-- <node type="throttle" name="depth_info_topic_throttle" pkg="topic_tools" required="true" output="screen"
        args="messages /camera/depth/camera_info $(arg image_throttle_rate) /camera/depth/camera_info_throttled"/> -->

    <!-- <node pkg="rosbag" type="record" name="rosbag_record_image"
       args="record -o $(find image_logger)/bags/camera_throttled
            /camera/color/image_raw_throttled
            /camera/aligned_depth_to_color/image_raw_throttled
            /camera/color/camera_info_throttled
            /camera/depth/camera_info_throttled
            " /> -->

    <node name="image_logger" pkg="image_logger" type="image_logger.py" output="screen" required="true">
        <!-- <param name="image_topic" value="/camera/color/image_raw_throttled"/> -->
        <param name="image_topic" value="/camera/aligned_depth_to_color/image_raw_throttled"/>
        <param name="keyboard_topic" value="/keys"/>
        <param name="save_dir" value="$(find image_logger)/images/"/>
    </node>
</launch>
